### ISSUE TITLE
__all__ can be a tuple or a list

### ISSUE THREAD

**Issue by smoyerx** [PyCQA/pydocstyle#62]
Parse_all() flags an error if **all** is not a tuple, whereas **all** is more commonly a list.

**Comment by ssbarnea** [comment]
Is there a way to disable these really annoying warnings? We have some legacy code that we don't want to update but we still have to maintain it for a while.

Why would we have a warning that cannot be disabled using the normal ignore mechanism?

**Comment by ask** [comment]
I'm not sure I see the point of this warning, as even `__all__` as a tuple can be modified:

```
import os

__all__ = ('g',)

def f():
    pass


def g():
    """Docstring."""


if os.environ.get('FEATURE_F'):
    __all__ += ('f',)
```

This makes __all__ point to a new tuple object in memory, so it's not actually mutable, but both lists and tuples have the same problem when statically parsing a module.

**Comment by jdufresne** [comment]
The [Python official docs](https://docs.python.org/3/tutorial/modules.html#importing-from-a-package) refer to `__all__` as a _list_ and write all examples using a list. This warning is contradicting the language's own documentation. Further, as @ask has shown, this offers no real protection against mutating the `__all__` reference.

IMO, I think you should consider removing thewarning and instead document pydocstyle's shortfall when handling modules that mutate `__all__`.
